### PR TITLE
Gate new profile-2 app

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Gate the profile-2 app to only allow logged-in users.
Also removed some old props.

## Testing done
Local

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/account/containers/AccountApp.jsx [hunk 1]
```diff
@@ -16,7 +16,6 @@ class AccountApp extends React.Component {
     return (
       <div>
         <RequiredLoginView
-          authRequired={1}
```
Comment (erikphansen) [inline src/applications/personalization/account/containers/AccountApp.jsx]: this prop is no longer supported by RequiredLoginView. It was removed here https://github.com/department-of-veterans-affairs/vets-website/commit/96d073c40b8282a24979b01713fa6050d1dede8b

File: src/applications/personalization/profile360/containers/VAProfileApp.jsx [hunk 2]
```diff
@@ -21,11 +21,8 @@ class VAProfileApp extends React.Component {
     return (
       <div>
         <RequiredLoginView
-          authRequired={1}
           serviceRequired={backendServices.USER_PROFILE}
           user={this.props.user}
-          loginUrl={this.props.loginUrl}
```
Comment (erikphansen) [inline src/applications/personalization/profile360/containers/VAProfileApp.jsx]: No longer supported by `RequiredLoginView`. These were removed here https://github.com/department-of-veterans-affairs/vets-website/pull/6961 and here https://github.com/department-of-veterans-affairs/vets-website/pull/6474

File: src/applications/personalization/profile-2/sass/profile-2.scss [hunk 3]
```diff
@@ -2,6 +2,9 @@
 @import "../../../../platform/forms/sass/m-schemaform";
```
Comment (bradl3yC) [inline src/applications/personalization/profile-2/sass/profile-2.scss]: can this import be switched to 'platform/forms/sass/m-schemaform' ?  or does sass not play well with those imports?

**Review by bradl3yC — APPROVED** [review]

File: src/applications/personalization/profile-2/sass/profile-2.scss (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/sass/profile-2.scss]: sadly no, Sass don't play that.

**Pull request merged**